tests/framework/utils.py

```diff
@@ -515,3 +516,26 @@ def compare_versions(first, second):
         return first[1] - second[1]
 
     return first[0] - second[0]
+
+
+class ResultsFileDumper:
+    """Class responsible with outputting test results to files."""
+
+    def __init__(self, test_name: str):
+        """Initialize the instance."""
+        self._root_path = defs.TEST_RESULTS_DIR
+        self._path = self._root_path / test_name
+
+        # Create the root directory, if it doesn't exist.
+        os.makedirs(self._root_path, exist_ok=True)
+
+    def writeln(self, data: str, append=True):
+        """Write the `data` string to the output file, appending a newline."""
+        if not append:
+            flags = "w"
+        else:
+            flags = "a"
+
+        with open(self._path, flags) as file:
```

Comment:
Why not open the file on class init and keep writing data as this function is called?

Reply:
that's a good point, we can close the file on `__del__`

Reply:
It's closed when it goes out of scope

Reply:
done